examples/wallet/app/actions/account.js

```diff
@@ -42,16 +47,73 @@ export function setAccount(privateKey: string): Thunk<SetKeysAction> {
   };
 }
 
-const initializeKeysAndRedirect = (dispatch, keys: AccountKeys) => {
+export function setAccountFromPrivateKey(
+  privateKey: string
+): Thunk<SetKeysAction> {
+  return function setAccountFromPrivateKeyThunk(dispatch) {
+    getAccountFromPrivateKey(privateKey)
+      .then(loadedPrivateKey => {
+        dispatch({
+          type: SET_KEYS,
+          ...loadedPrivateKey
+        });
+        return dispatch(updateAccountTransactionsAndState());
+      })
+      .catch(error => {
+        console.log(error);
+        throw new AccountsException(
+          PRIVATE_KEY_ERROR,
+          'There was an error settings loading private key'
+        );
+      });
+  };
+}
+
+export function updateAccountTransactionsAndState(): Thunk<SetKeysAction> {
+  return function updateAccountTransactionsAndStateThunk(dispatch) {
+    return Promise.all([
+      dispatch(updateAccountTransactions()),
+      dispatch(updateNodeSettings()),
+      dispatch(updateAccountState())
+    ])
+      .then(() => dispatch(push(routes.WALLET)))
+      .catch(error => {
+        console.log(error);
+        throw new AccountsException(
+          ACCOUNT_STATE_ERROR,
+          'There was an error retrieving balance, transactions and node settings'
+        );
+      });
+  };
+}
+
+function AccountsException(errorType: string, message: string) {
```

Comment:
It'd be more idiomatic to throw an Error instance with the constant as the message.
If you want you could also have severay classes of errors (It's also pretty idiomatic to differentiate between exceptions by class). These'll be the first errors being thrown around the app, and later we'll want to be consistent, so we'll follow what you define here.

Reply:
trhow Error